pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -11580,7 +11581,51 @@ spec:
 		t.Fatal("fail to marshal task", err)
 	}
 
-	prs := parse.MustParseV1beta1PipelineRun(t, fmt.Sprintf(`
+	noMatchPolicy := []*v1alpha1.VerificationPolicy{{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "no-match",
+			Namespace: ts.Namespace,
+		},
+		Spec: v1alpha1.VerificationPolicySpec{
+			Resources: []v1alpha1.ResourcePattern{{Pattern: "no-match"}},
+		}}}
+	// warnPolicy doesn't contain keys so it will fail verification but doesn't fail the run
+	warnPolicy := []*v1alpha1.VerificationPolicy{{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "warn-policy",
+			Namespace: ts.Namespace,
+		},
+		Spec: v1alpha1.VerificationPolicySpec{
+			Resources: []v1alpha1.ResourcePattern{{Pattern: ".*"}},
+			Mode:      v1alpha1.ModeWarn,
+		}}}
+
+	testCases := []struct {
+		name                 string
+		task                 []*v1beta1.Task
+		noMatchPolicy        string
+		verificationPolicies []*v1alpha1.VerificationPolicy
+	}{{
+		name:                 "ignore no match policy",
+		noMatchPolicy:        config.IgnoreNoMatchPolicy,
+		verificationPolicies: noMatchPolicy,
+	}, {
+		name:                 "warn no match policy",
+		noMatchPolicy:        config.WarnNoMatchPolicy,
+		verificationPolicies: noMatchPolicy,
+	}, {
+		name:                 "pass enforce policy",
+		noMatchPolicy:        config.FailNoMatchPolicy,
+		verificationPolicies: vps,
+	}, {
+		name:                 "only fail warn policy",
+		noMatchPolicy:        config.FailNoMatchPolicy,
+		verificationPolicies: warnPolicy,
+	},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			prs := parse.MustParseV1beta1PipelineRun(t, fmt.Sprintf(`
 metadata:
```

Comment:
The `prs` seems not to change for different test cases. Can we define it outside of the for loop?

Reply:
ah I intentionally moved it inside the loop before cuz there are some issues, but now it passes. So moved it out of the loop.